src/canisters/registrar/src/service.rs

```diff
@@ -1288,6 +1335,139 @@ impl RegisterCoreContext {
     }
 }
 
+pub struct RegistrationNameQueryContext<'a> {
+    token_index_store: &'a TokenIndexStore,
+    registration_store: &'a RegistrationStore,
+}
+
+impl<'a> RegistrationNameQueryContext<'a> {
+    pub fn new(
+        token_index_store: &'a TokenIndexStore,
+        registration_store: &'a RegistrationStore,
+    ) -> Self {
+        Self {
+            token_index_store: token_index_store,
+            registration_store: registration_store,
+        }
+    }
+
+    pub fn get_all_unexpired_registrations(&self, now: u64) -> Vec<UnexpiredRegistrationAggDto> {
```

Comment:
Use the map, filter to replace the `for in` implementation to keep it consistent with your previous style